src/validation.ts

```diff
@@ -1,57 +1,97 @@
-import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import { HttpHandler, HttpRequest, InvocationContext } from '@azure/functions';
 import { AnySchema } from 'joi';
 
 import { ApplicationError } from './error';
+import { BeforeExecutionFunction, MiddlewareResult, PostExecutionFunction, isErrorResult } from './middleware';
+import { stringify } from './util/stringify';
 
 type ValidationOptions = Partial<{
     transformErrorMessage: (message: string) => unknown;
-    extractValidationContentFromRequest: (context: Context, req: HttpRequest) => unknown;
+    extractValidationContentFromRequest: (
+        req: HttpRequest,
+        context: InvocationContext,
+        result: MiddlewareResult<ReturnType<HttpHandler>>,
+    ) => unknown;
     shouldThrowOnValidationError: boolean;
+    skipIfResultIsFaulty: boolean;
+    printRequest: boolean;
 }>;
 
-export function requestValidation(schema: AnySchema, opts?: ValidationOptions): AzureFunction {
-    return (context: Context, req: HttpRequest): Promise<void> => {
-        context.log.verbose('Validating the request.');
+export function requestValidation(schema: AnySchema, opts?: ValidationOptions): BeforeExecutionFunction<HttpHandler> {
+    const skipIfResultIsFaulty = opts?.skipIfResultIsFaulty ?? true;
+    const printRequest = opts?.printRequest ?? false;
 
-        const toBeValidatedContent = opts?.extractValidationContentFromRequest?.(context, req) ?? req.body;
-        const shouldThrowOnValidationError = opts?.shouldThrowOnValidationError ?? true;
+    return async (req, context, result): Promise<void> => {
+        if (skipIfResultIsFaulty && result.$failed) {
+            context.info('Skipping request-validation because the result is faulty.');
+            return Promise.resolve();
+        }
 
-        const validationResult = schema.validate(toBeValidatedContent);
-        if (validationResult && validationResult.error) {
-            context.log.error('The request did not match the given schema.');
-            context.log.verbose(validationResult);
+        const clonedRequest = req.clone(); //see https://github.com/nuxt/nuxt/issues/19245
+        context.info('Validating the request.');
+        try {
+            const toBeValidatedContent =
+                opts?.extractValidationContentFromRequest?.(clonedRequest, context, result) ??
+                (await clonedRequest.json());
+            const shouldThrowOnValidationError = opts?.shouldThrowOnValidationError ?? true;
 
-            if (shouldThrowOnValidationError) {
+            const validationResult = schema.validate(toBeValidatedContent);
+            if (validationResult && validationResult.error) {
+                context.error(
```

Comment:
Ich würde das `validationResult` gleich in der Meldung mitloggen, dann findet man es beim Durchsuchen der Logs schneller